diff --git a/node_modules/@shopify/flash-list/dist/recyclerview/RecyclerView.js b/node_modules/@shopify/flash-list/dist/recyclerview/RecyclerView.js
index 4324707..d6bc120 100644
--- a/node_modules/@shopify/flash-list/dist/recyclerview/RecyclerView.js
+++ b/node_modules/@shopify/flash-list/dist/recyclerview/RecyclerView.js
@@ -83,9 +83,11 @@ const RecyclerViewComponent = (props, ref) => {
             containerViewSizeRef.current = outerViewSize;
             // firstChildViewLayout is already relative to the outer container,
             // so its x/y directly gives the first item offset.
-            const firstItemOffset = horizontal
-                ? firstChildViewLayout.x
-                : firstChildViewLayout.y;
+            const firstItemOffset = inverted
+                ? 0
+                : horizontal
+                    ? firstChildViewLayout.x
+                    : firstChildViewLayout.y;
             // Update the RecyclerView manager with window dimensions
             recyclerViewManager.updateLayoutParams({
                 width: horizontal ? outerViewSize.width : firstChildViewLayout.width,
diff --git a/node_modules/@shopify/flash-list/src/recyclerview/RecyclerView.tsx b/node_modules/@shopify/flash-list/src/recyclerview/RecyclerView.tsx
index 3c25e7e..0525156 100644
--- a/node_modules/@shopify/flash-list/src/recyclerview/RecyclerView.tsx
+++ b/node_modules/@shopify/flash-list/src/recyclerview/RecyclerView.tsx
@@ -176,9 +176,11 @@ const RecyclerViewComponent = <T,>(
 
       // firstChildViewLayout is already relative to the outer container,
       // so its x/y directly gives the first item offset.
-      const firstItemOffset = horizontal
-        ? firstChildViewLayout.x
-        : firstChildViewLayout.y;
+      const firstItemOffset = inverted
+        ? 0
+        : horizontal
+          ? firstChildViewLayout.x
+          : firstChildViewLayout.y;
 
       // Update the RecyclerView manager with window dimensions
       recyclerViewManager.updateLayoutParams(
